dimod/binary_quadratic_model.py

```diff
@@ -1612,6 +1612,92 @@ def from_json(cls, obj):
 
         return json.load(obj,  object_hook=bqm_decode_hook)
 
+    def to_bson(self, fp=None):
+        """Convert a binary quadratic model to an effecient BSON format.
+
+        Args:
+            fp (file, optional):
+                A `.write()`-supporting `file object`_ to save the binary 
+                quadratic model encoded in BSON format to. If not provided, 
+                returns bytes.
+        """
+                
+
+        import bson
```

Comment:
Should we add `bson` dep to `all` extras?

Reply:
yes it should be added.